process/src/services/service.ports.ts

```diff
@@ -1,295 +1,406 @@
-        return this._unrefPortListeners(session, port);
-    }
+  public unrefPort(session: string, port: string): Promise<void> {
+    return this._unrefPortListeners(session, port);
+  }
 
-    public write(port: string, chunk: Buffer | string): Promise<void> {
-        return new Promise((resolve, reject) => {
-            let controller: ControllerSerialPort | undefined = this._controllers.get(port);
-            if (controller === undefined) {
-                return reject(new Error(this._logger.error(`Fail to write into port, because controller of port "${port}" isn't created.`)));
-            }
-            controller.write(chunk).then(resolve).catch(reject);
-        });
-    }
+  public write(port: string, chunk: Buffer | string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      let controller: ControllerSerialPort | undefined =
+        this._controllers.get(port);
+      if (controller === undefined) {
+        return reject(
+          new Error(
+            this._logger.error(
+              `Fail to write into port, because controller of port "${port}" isn't created.`
+            )
+          )
+        );
+      }
+      controller.write(chunk).then(resolve).catch(reject);
+    });
+  }
 
-    public getList(): Promise<IPortInfo[]> {
-        return new Promise((resolve, reject) => {
-            SerialPort.list().then((ports: SerialPort.PortInfo[]) => {
-                const portsU = ports as unknown;
-                const portsI: IPortInfo[] = portsU as IPortInfo[];
-                resolve(portsI.map((port: IPortInfo) => {
-                    return Object.assign({ }, port);
-                }));
-            }).catch((error: Error | null | undefined) => {
-                if (error) {
-                    reject(new Error(this._logger.error(`Fail to get list of ports due error: ${error.message}`)));
-                }
-                reject(new Error(this._logger.error(`Fail to get list of ports because unknown error`)));
-            });
+  public getList(): Promise<IPortInfo[]> {
+    return new Promise((resolve, reject) => {
+      SerialPort.list()
+        .then((ports: SerialPort.PortInfo[]) => {
+          const portsU = ports as unknown;
+          const portsI: IPortInfo[] = portsU as IPortInfo[];
+          resolve(
+            portsI.map((port: IPortInfo) => {
+              return Object.assign({}, port);
+            })
+          );
+        })
+        .catch((error: Error | null | undefined) => {
+          if (error) {
+            return reject(
+              new Error(
+                this._logger.error(
+                  `Fail to get list of ports due error: ${error.message}`
+                )
+              )
+            );
+          }
+          reject(
+            new Error(
+              this._logger.error(
+                `Fail to get list of ports due to an unknown error`
+              )
+            )
+          );
```

Comment:
maybe just
```
reject(
            new Error(
              this._logger.error(
                `Fail to get list of ports because of: ${error ? error.message : 'an unknown error'}`
              )
            )
          );
```